src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -234,14 +237,36 @@ close()
 {
     if(! strand_.running_in_this_thread())
         return strand_.post(std::bind(
-            &BaseWSPeer::close, impl().shared_from_this()));
+            static_cast<void(BaseWSPeer<Handler, Impl>::*)(void)>(&BaseWSPeer::close),
```

Comment:
I suggest a clarifying comment for the need for this case, and a slight simplification in its syntax:
```
            // disambiguate to this close overload
            static_cast<void(BaseWSPeer::*)()>(&BaseWSPeer::close),
```